viz.Rmd

```diff
@@ -671,35 +1012,67 @@ vector graphics as much as you like without the image looking bad, while raster
+to save a file containing this image. 
+`ggsave` works by taking a file name to create for the image 
+as it's first argument. 
+This can include the path to the directory where you would like to save the file 
+(e.g., `img/filename.png` to save a file named `filename` to the `img` directory),
+and the name of the plot object to save as it's second argument.
+The kind of image to save is specified by the file name's file extension.
+For example, 
+to create a PNG image file we specify that the file extension is `.png`.
+Below we demonstrate how to save PNG, JPG, BMP, TIFF and SVG file types 
+for the `faithful_plot`:
+
+```{r warning=FALSE, message=FALSE}
 ggsave("img/faithful_plot.png", faithful_plot)
 ggsave("img/faithful_plot.jpg", faithful_plot)
 ggsave("img/faithful_plot.bmp", faithful_plot)
 ggsave("img/faithful_plot.tiff", faithful_plot)
 ggsave("img/faithful_plot.svg", faithful_plot)
+```
 
-print(paste("PNG filesize: ", file.info("img/faithful_plot.png")["size"] / 1000000, "MB"))
-print(paste("JPG filesize: ", file.info("img/faithful_plot.jpg")["size"] / 1000000, "MB"))
-print(paste("BMP filesize: ", file.info("img/faithful_plot.bmp")["size"] / 1000000, "MB"))
-print(paste("TIFF filesize: ", file.info("img/faithful_plot.tiff")["size"] / 1000000, "MB"))
-print(paste("SVG filesize: ", file.info("img/faithful_plot.svg")["size"] / 1000000, "MB"))
+```{r, filesizes, echo = FALSE}
+file_sizes <- tibble(`Image type` = c("Bitmap / Raster", 
+                        "Bitmap / Raster", 
+                        "Bitmap / Raster", 
+                        "Bitmap / Raster",
+                        "Vector / Scalable Graphics"),
+       `File type` = c("PNG", "JPG", "BMP", "TIFF", "SVG"),
+       `Image size` = c(paste(round(file.info("img/faithful_plot.png")["size"] / 1000000, 2), "MB"),
+                        paste(round(file.info("img/faithful_plot.jpg")["size"] / 1000000, 2), "MB"),
+                        paste(round(file.info("img/faithful_plot.bmp")["size"] / 1000000, 2), "MB"),
+                        paste(round(file.info("img/faithful_plot.tiff")["size"] / 1000000, 2), "MB"),
+                        paste(round(file.info("img/faithful_plot.svg")["size"] / 1000000, 2), "MB")))
+kable(file_sizes,
+      caption = "File sizes of `faithful_plot` when saved as different file formats.")
 ```
+
+The different file types are not only different in their properties 
+(outlined above), but also in their file sizes (Table \@ref(tab:filesizes)).
 Wow, that's quite a difference! Notice that for such a simple plot with few
 graphical elements (points), the vector graphics format (SVG) is over 100 times
 smaller than the uncompressed raster images (BMP, TIFF). Also, note that the
 JPG format is twice as large as the PNG format since the JPG compression
-algorithm is designed for natural images (not plots). Below, we also show what
+algorithm is designed for natural images (not plots). 
+
+Below, we also show what
```

Comment:
images are stacked in the pdf instead of side by side -- they also look the same... 
<img width="586" alt="Screen Shot 2021-10-01 at 1 33 23 PM" src="https://user-images.githubusercontent.com/3838091/135683021-85e16495-4cd3-4d77-ab11-fc715fe1a220.png">

Reply:
Yep, aware and will fix when we attack the PDF